src/components/game/game.js

```diff
@@ -0,0 +1,262 @@
+import React, { Component } from 'react';
+import Board from '../board';
+import Header from '../header';
+import Footer from '../footer';
+import Settings from '../settings';
+import Overlay from '../overlay';
+
+import './game.css';
+
+import moveSound from './sounds/move-sound.mp3';
+
+import shuffleArr from './methods/shuffleArr';
+import createData from './methods/createData';
+
+export default class Game extends Component {
+  constructor() {
+    super();
+    this.startBoardSize = 18;
+    this.startSettings = {cardBack: 'red', boardSize: 18, cardType: 'pictures'}
+    
+    this.state = {
+      data: shuffleArr(createData(this.startSettings.boardSize, this.startSettings.cardType)),
+      clickCounter: 0,
+      movesCounter: 0,
+      isWin: false,
+      isSettingsOpen: false,
+      isGameStart: false,
+      choosenSettings: {
```

Comment:
знаешь, спред опреатор deep copy объекта не делает. Поэтому здесь можно просто приравнять:
```suggestion
      choosenSettings: this.startSettings
```